src/Interfaces/RepositoryInterface.php

```diff
@@ -4,20 +4,23 @@
 
 namespace Swis\JsonApi\Client\Interfaces;
 
+/**
+ * @template TItem of \Swis\JsonApi\Client\Interfaces\ItemInterface
+ */
 interface RepositoryInterface
 {
     /**
-     * @return \Swis\JsonApi\Client\Interfaces\CollectionDocumentInterface
+     * @return \Swis\JsonApi\Client\Interfaces\CollectionDocumentInterface<TItem>
      */
     public function all();
 
     /**
-     * @return \Swis\JsonApi\Client\Interfaces\ItemDocumentInterface
+     * @return \Swis\JsonApi\Client\Interfaces\ItemDocumentInterface<TItem>
      */
     public function find(string $id);
```

Comment:
WHen i check the tests, i think this return type is not correct. It seems this can also return a DocumentInterface. See `RepositoryTest::it_can_find_one`.

This could also mean the interface on the Document is wrong. seems a little weird, since ItemDocument is pretty much the same class with the extra getData method.

Think there is something wrong here. @JaZo any insights?

Reply:
Yeah I had the same questions, but I didnt want to touch actual code in this PR…

Reply:
I would probably make it documentinterface, since itemdocumentinterface does extend documentinterface. But that will mean some errors later with phpstan when we are calling getData.

Reply:
It would also make this PR mostly obsolete if find() would not resolve to an Item 😋

Reply:
The primary reason to have a `CollectionDocumentInterface` and an `ItemDocumentInterface` is to differentiate between the return types of `getData`. The former returns a collection and the latter an item. This stems from ancient times when generics weren't a thing. I'm okay with changing that to generics, but that would probably be a breaking change, so maybe we should leave that for now and just add a todo.

Reply:
I’m fine with that